pkgs/os-specific/darwin/apple-source-releases/xnu/default.nix

```diff
@@ -62,18 +69,25 @@ appleDerivation' (if headersOnly then stdenvNoCC else stdenv) ({
   HOST_BISON = "bison";
   HOST_GM4 = "m4";
   MIGCC = "cc";
-  ARCHS = "x86_64";
+  ARCHS = if stdenv.isx86_64 then "x86_64" else "arm64";
+  ARCH_CONFIGS = if stdenv.isx86_64 then "x86_64" else "arm64";
+
+  # fails on aarch64-darwin because of missing i386 headers
+  doInstallCheck = stdenv.isx86_64;
```

Comment:
Are you disabling our header list check here? Can we just add another list for arm64?

Reply:
Nice catch. Let me try this approach.

Reply:
A new list for arm64 is added.